Updates to solver-related comments

domainexpert opened the pull request.

delcypher commented:
Issues.

- The name `computeValidity()` is confusing because it doesn't really compute validity (`computeTruth()` does that). It instead is computing feasibility of the paths from a two way branch which can be implemented by multiple checks of validity.

- `computeTruth()` actually computes validity.

- We have comments that says things like 

```
Determine whether the given query expression is provably true given the constraints.
```

This is awful. "provably true" means very little to me. I think the approach taken in https://github.com/klee/klee/blob/master/include/klee/SolverImpl.h which gives the logical formula is helpful (given both as a validity query and as a satisfiability query). That's not enough the language needs to be clearer.

domainexpert commented:
@delcypher I recently had a long email and chat discussions on the functionality of those functions. The names are bound to confuse since "validity" and "truth" are synonyms. I suggest keeping the name `computeTruth` as is to minimize change, but change `computeValidity` to `computeTruthOrFalsity`. For this, I can add a commit to this PR.

The usage of the word `provably` in the sentence that you mentioned seems superfluous. I suggest removal of the word `provably` in all member function comments. For `TimingSolver::evaluate`, I also suggest the removal of `or neither`. I can add another commit for this.

I'm happy to try to complete the comments for the functions of `IncompleteSolver` in the style of `SolverImpl`. I can also add another commit for this.

domainexpert commented:
As promised, these are the commits, but I did not follow what I promised to the letter. So what I did:
1. Instead of renaming `computeValidity` member functions to `computeTruthOrFalsity`, I renamed them to `computeValidityMode`, which seems more appropriate. I also renamed `Solver::Validity` to `Solver::ValidityMode` and `Solver::PartialValidity` to `Solver::PartialValidityMode`. 
2. I did not remove all usages of 'provably' from all comments, instead I changed their usages, based on the understanding that what it means is that 'it is can be shown by KLEE (and the underlying solver)'.
3. Extended member function comments for `IncompleteSolver::partialValidityMode` and `IncompleteSolver::computeTruth` in the style of `SolverImpl`.
